src/subscribables/dependentObservable.js

```diff
@@ -250,23 +242,62 @@ ko.computed = ko.dependentObservable = function (evaluatorFunctionOrOptions, eva
         pure = true;
         isSleeping = true;     // Starts off sleeping; will awake on the first subscription
         dependentObservable.beforeSubscriptionAdd = function (event) {
-            // If asleep, wake up the computed and evaluate to register any dependencies.
-            if (event == 'change' && isSleeping) {
+            // If asleep, wake up the computed by subscribing to any dependencies.
+            if (!_isDisposed && isSleeping && event == 'change') {
                 isSleeping = false;
-                _needsEvaluation = true;
-                evaluateImmediate(true /* suppressChangeNotification */);
-                if (!_isDisposed) {     // test since evaluateImmediate could trigger disposal
+                if (_needsEvaluation || haveDependenciesChanged()) {
+                    dependencyTracking = null;
+                    _dependenciesCount = 0;
+                    _needsEvaluation = true;
+                    evaluateImmediate();
+                } else {
+                    // First put the dependencies in order
```

Comment:
I'm sure this manual order management is necessary, but I'm struggling a bit to understand why so (or what's different from before when we didn't need this). Could you clarify?

Also I don't see anything obvious in the added test that shows anything to do with dependency order manangement. In general the amount of test changes are quite small compared with the amount of change to the production code. Is this because our existing tests already demonstrate all the functionality added here, or would it be worth ensuring more of the cases accounted for in the code here are shown in the tests?